src/main/java/io/github/thunderz99/cosmos/util/JsonUtil.java

```diff
@@ -409,88 +407,112 @@ public static <T> T fromJson(InputStream is, String className) {
-			processException(map, e);
-		}
-		return null;
-	}
-
-	private static void processException(Object object, Exception e) {
-		throw new IllegalArgumentException("json process error.", e);
-	}
-
-	private static ObjectMapper newObjectMapper() {
-		ObjectMapper mapper = new ObjectMapper();
-		init(mapper);
-		return mapper;
-	}
-
-	/**
-	 * return an objectMapper configured the same as this JsonUtil
-	 * @return objectMapper
-	 */
-	public static ObjectMapper getObjectMapper() {
-		return newObjectMapper();
-	}
+    /**
+     * Json to List of bean
+     *
+     * @param is        json string inputStream
+     * @param className className for bean
+     * @param <T>       generic param for bean
+     * @return List of bean
+     */
+    public static <T> List<T> fromJson2List(InputStream is, String className) {
+        try {
+            JavaType javaType = constructListType(className);
+            return mapper.readValue(is, javaType);
+        } catch (Exception e) {
+            processException(is, e);
+        }
+        return List.of();
+    }
+
+    /**
+     * Json to List of bean
+     *
+     * @param is       json string inputStream
+     * @param classOfT class of bean
+     * @param <T>      generic param for bean
+     * @return List of bean
+     */
+    public static <T> List<T> fromJson2List(InputStream is, Class<T> classOfT) {
+        try {
+            JavaType javaType = constructListType(classOfT.getName());
+            return mapper.readValue(is, javaType);
+        } catch (Exception e) {
+            processException(is, e);
+        }
+        return List.of();
+    }
+
+    /**
+     * Json string to bean object
+     *
+     * @param is       json string inputStream
+     * @param javaType javaType for bean
+     * @param <T>      generic for bean
+     * @return bean object
+     */
+    public static <T> T fromJson(InputStream is, JavaType javaType) {
+        try {
+            return mapper.readValue(is, javaType);
+        } catch (Exception e) {
+            processException(is, e);
+        }
+        return null;
+    }
+
+    /**
+     * map to bean object
+     *
+     * @param map      map representing json
+     * @param classOfT class of bean
+     * @param <T>      generic param for bean
+     * @return bean object
+     */
+    public static <T> T fromMap(Map<String, Object> map, Class<T> classOfT) {
+        try {
+            return mapper.convertValue(map, classOfT);
+        } catch (Exception e) {
+            processException(map, e);
+        }
+        return null;
+    }
+
+    private static void processException(Object object, Exception e) {
+        log.error("json process error. e = \n{}", exceptionToString(e));
+        if (Modifier.isFinal(object.getClass().getModifiers())) {
+            log.error("json process error. object = {}", object);
+        } else {
+            log.error("json process error. object = {}", ToStringBuilder.reflectionToString(object));
```

Comment:
Please change log level from ERROR to WARN. Because the library should not decide whether this exception should be reported as an ERROR. 
We should let this decision done by the user application. 
We already throw this error as an IllegalArgumentException, so the user application can log this exception as an ERROR.